tests/test_unit_security.py

```diff
@@ -0,0 +1,50 @@
+from sqlmodel import select
+
+from fractal_server.app.models.security import UserOAuth
+from fractal_server.app.security import _create_first_user
+
+
+async def count_users(db):
+    res = await db.execute(select(UserOAuth))
+    return len(res.unique().scalars().all())
+
+
+async def test_unit_create_first_user(db):
+
+    assert await count_users(db) == 0
+
+    await _create_first_user(email="test1@fractal.com", password="xxxx")
+    assert await count_users(db) == 1
+
+    await _create_first_user(
+        email="test2@fractal.com", password="xxxx", username="test2"
+    )
+    assert await count_users(db) == 2
+
+    # UserAlreadyExists
+    await _create_first_user(email="test2@fractal.com", password="xxxx")
```

Comment:
To test logs, one should use the `caplog` fixture (see e.g. https://docs.pytest.org/en/7.1.x/how-to/logging.html).

Reply:
See also https://github.com/search?q=repo%3Afractal-analytics-platform%2Ffractal-server%20caplog&type=code